geoclustering/kmeans.py

```diff
@@ -39,9 +35,17 @@ def __init__(self, Z, row_clusters, col_clusters, n_row_clusters,
         self.kmean_n_clusters = kmean_n_clusters
         self.kmean_max_iter = kmean_max_iter
 
-        if len(np.unique(row_clusters)) > n_row_clusters or \
-                len(np.unique(col_clusters)) > n_col_clusters:
-            raise ValueError
+        if len(np.unique(row_clusters)) > n_row_clusters:
+            print('Setting "n_row_clusters" to {}, \
+            accoding to the number of unique elements in "row_clusters".'
+                  .format(len(np.unique(row_clusters))))
+            n_row_clusters = len(np.unique(row_clusters))
+
+        if len(np.unique(col_clusters)) > n_col_clusters:
+            print('Setting "col_clusters" to {}, \
+            accoding to the number of unique elements in "col_clusters".'
+                  .format(len(np.unique(col_clusters))))
+            n_col_clusters = len(np.unique(col_clusters))
```

Comment:
Still concerning this: does it makes sense to have the possibility of setting `n_col_clusters` and `n_row_clusters` from input? Shoulnd't they exactly match `np.unique(col_clusters)` and `np.unique(row_clusters)` - or, related,  what happens if ones sets an initial number of clusters that differs from these values?

Reply:
To summarize the discussion with Francesco this morning: this implementation intends to keep the information of missing classes in case some classes are not selected in the co-clustering. At this stage it is hard to estimate the consequences of discarding this information, so we decide to keep it for now.